### ISSUE TITLE
S3 spy raises NoMethodError if bucket name is a symbol

### ISSUE THREAD

**Issue by eric-christian** [elastic/apm-agent-ruby#995]
## Describe the bug

S3 spy raises NoMethodError if bucket name is a symbol.

```
Failure/Error: s3_client.get_object(bucket: BUCKET_NAME, key: export.id)
NoMethodError:
  undefined method `rindex' for :"withdrawal.exports":Symbol
# /bundler/accounting_service_2.7.2/ruby/2.7.0/gems/elastic-apm-3.15.0/lib/elastic_apm/spies/s3.rb:45:in `bucket_name'
```

## Expected behavior

Convert the symbol to a string and don't raise an exception.

Since the documentation from the s3 client states that the bucket name should be a "String", i'm not entirely sure if this can be considered a bug at all.
`:bucket (required, String) — The bucket name containing the object.`
One could say that we use the s3 client in a wrong way and that would be true.
However, the s3 client works well with a symbol as bucket name. So if the spy is supposed to be minimal invasive, an additional `.to_s` can't be wrong.

Maybe? Maybe not? I don't know. I just wanted to let you know. :)

Have a great day!
Eric

## Environment

- **Ruby version**: 2.7.2
- **Agent version**: 3.15.0

**Comment by mikker** [comment]
I agree this is a bug. Will fix on Tuesday when I'm back from Easter holiday. Until then perhaps you can do the `.to_s` in your code. Thanks for reporting!

**Comment by mikker** [comment]
The fix was just merged into master, the upcoming 4.0.0 release, and also included in 3.15.1.

**Comment by mikker** [comment]
Thank you for reporting this!
